core/src/main/java/org/apache/seata/core/store/db/sql/lock/SqlServerLockStoreSql.java

```diff
@@ -31,8 +31,19 @@ public class SqlServerLockStoreSql extends AbstractLockStoreSql {
     private static final String INSERT_LOCK_SQL_SQLSERVER = "insert into " + LOCK_TABLE_PLACE_HOLD + "(" + ALL_COLUMNS + ")"
             + " values (?, ?, ?, ?, ?, ?, ?, SYSDATETIME(), SYSDATETIME(), ?)";
 
+    /**
+     * The constant QUERY_ALL_LOCK.
+     */
+    private static final String QUERY_ALL_LOCK_SQLSERVER = "select " + ALL_COLUMNS + " from " + LOCK_TABLE_PLACE_HOLD
```

Comment:
For some type of database, like SQLServer, it has unique SQL grammer. Special care must be taken in handling their order-by conditions. So I extract the order-by condition, just like the where condition.

Reply:
> For some type of database, like SQLServer, it has unique SQL grammer. Special care must be taken in handling their order-by conditions. So I extract the order-by condition, just like the where condition.

但是我看到你从头到尾只使用了`gmt_create desc`，所以为什么它不直接放入到`QUERY_ALL_LOCK_SQLSERVER`中？因为实际上直接放入`QUERY_ALL_LOCK_SQLSERVER`中，也已经约束了他的order-by condition了
But I noticed that you only used gmt_create desc from beginning to end, so why doesn't it directly put it into `QUERY_ALL_LOCK_SQLSERVER`? Because in fact, directly putting it into `QUERY_ALL_LOCK_SQLSERVER` has already constrained its order-by condition

Reply:
Well, I suppose that it is a SQL template. Other order-by conditions, like "a desc b asc", not only "gmt_create desc"  may be used later. It is not a good idea to make it fixed.

Reply:
> Well, I suppose that it is a SQL template. Other order-by conditions, like "a desc b asc", not only "gmt_create desc" may be used later. It is not a good idea to make it fixed.

I understand that you're considering future extensibility, but currently, your approach is limited to SQLServer and doesn't effectively enable other types of database implementations.  Therefore, I believe the priority should be to ensure consistency across multiple database types first. If extensibility for other databases is required later on, all database implementations should be addressed simultaneously. Personally, I find your current approach of exclusively handling SQLServer less elegant.

Reply:
As you expected, I have made the corrections as requested. Please review, thank you!
![image](https://github.com/apache/incubator-seata/assets/55011193/78b1aa37-1a43-4750-8cae-e9358737d1f7)
![image](https://github.com/apache/incubator-seata/assets/55011193/808020d7-367c-46a0-8d67-8bf37f383cd5)
![image](https://github.com/apache/incubator-seata/assets/55011193/fe632041-7889-415f-a703-f40dd722dc19)

Reply:
@funky-eyes I wonder whether you can receive the notifications email if I don't @ you. Pleas help review! Thanks a billon!

Reply:
> @funky-eyes I wonder whether you can receive the notifications email if I don't @ you. Pleas help review! Thanks a billon!

I can receive emails whenever you leave a message in the Seata community, whether it's an issue or a pull request, even if you don't mention me.